src/lib.rs

```diff
@@ -86,6 +86,18 @@ impl<T> LazyCell<T> {
         unsafe { &*self.inner.get() }.as_ref()
     }
 
+    /// Borrows the contents of this lazy cell mutably for the duration of the cell
+    /// itself.
+    ///
+    /// This function will return `Some` if the cell has been previously
+    /// initialized, and `None` if it has not yet been initialized.
+    ///
+    /// `borrow_mut` can be used to manually fill and reset a `LazyCell`
+    /// if you have exclusive access to it.
+    pub fn borrow_mut(&mut self) -> &mut Option<T> {
```

Comment:
Question: I really hate how this is not symmetric with `borrow`: `&mut Option<T>` vs `Option<&T>`. But we need `&mut Option`, because otherwise one won't be able to reset an already filled cell. Perhaps a better alternative would be to add two methods?

```
fn reset(&mut self) (-> bool)?
fn borrow_mut(&mut self) -> Option<&mut T>
```

Reply:
Wait, `reset` is basically `*lazycell = LazyCell::new()`, so we can just unify the API without any additions and loss of generality.